src/commands/createRelease.js

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env node
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 Open Assessment Technologies SA;
+ */
+
+const log = require('../log.js');
+
+const commander = require('commander');
+const program = new commander.Command();
+
+program
+    .name('taoRelease createRelease')
+    .usage('[options]')
+    .option('-d, --debug', 'output extra debugging')
+    // options with defaults
+    .option('--base-branch <branch>', 'the source branch for the release', 'develop')
+    .option('--branch-prefix <prefix>', 'the prefix of the branch created for releasing', 'release')
+    .option('--origin <remotename>', 'the name of the remote repo', 'origin')
+    .option('--release-branch <branch>', 'the target branch for the release PR', 'master')
+    .option('--www-user <user>', 'the user who runs php commands', 'www-data')
+    // options which fall back to user prompts if undefined
+    .option('--path-to-tao <path>', 'path to local TAO instance')
+    .option('--extension-to-release <extension>', 'camelCase name of the extension to release')
+    .option('--version-to-release <version>', 'version of the extension to release')
+    .option('--update-translations', 'indicates if we need to update translations')
+    .option('--release-comment <comment>', 'comment to add to github release')
+    .parse(process.argv);
```

Comment:
I thought about that, but decided that it would just be one more level of unnecessary abstraction. The available options have to be different for the 3 commands, and the concerned files are really easy to find within the project. So I duplicated that part and added/removed lines, expecting that we probably aren't going to update it often, and if we do, it's trivial to keep the 3 places in sync (and also low-impact in terms of mistakes).

Reply:
I not fully agree. Even if some options are specific to one command, most of them are common and should remain common. This is part of the API consistency.

The duplication is not a blocker for now, but even if it may look trivial today, it might introduce issues later.

Reply:
Ok. How about if I centralise them like so:

cliOptions.js
```
// the option values must be an array with the format ['--option-name <value>', 'description', 'default']
// @see https://www.npmjs.com/package/commander#options
module.exports = {
    baseBranch: ['--base-branch <branch>', 'the source branch for the release', 'develop']
};
```

prepareRelease.js
```
const cliOptions =  require('./cliOptions');
program
    .option(...cliOptions.baseBranch)

```

Reply:
@jsconan  I pushed the above idea, please take a look.

Reply:
Yes, looks good